pkg/libinit/proc_linux.go

```diff
@@ -31,6 +29,39 @@ func WaitOrphans() uint {
 	return numReaped
 }
 
+// WithTTYControl turns on controlling the TTY on this command.
+func WithTTYControl(ctty bool) CommandModifier {
+	return func(c *exec.Cmd) {
+		if c.SysProcAttr == nil {
+			c.SysProcAttr = &syscall.SysProcAttr{}
+		}
+		c.SysProcAttr.Setctty = ctty
+		c.SysProcAttr.Setsid = ctty
+	}
+}
+
+// WithCloneFlags adds clone(2) flags to the *exec.Cmd.
+func WithCloneFlags(flags uintptr) CommandModifier {
+	return func(c *exec.Cmd) {
+		if c.SysProcAttr == nil {
+			c.SysProcAttr = &syscall.SysProcAttr{}
+		}
+		c.SysProcAttr.Cloneflags = flags
+	}
+}
+
+func init() {
```

Comment:
I did it this way so I don't have to create an osDefault for every possible kernel. Suggestions welcome.

Reply:
Define osDefault globally, assign only for Linux and plan9, and then check if osDefault != nil before using it?

Reply:
I just went with a function. It will be ok.